RIOT-OS/RIOT
# chacha20poly1305: Initial implementation of AEAD cipher.

### Contribution description

This PR adds the chacha20poly1305 AEAD symmetric key cipher algorithm to RIOT-os. 

I didn't reuse the existing generic chacha code. There are some issues with that code that prevent reuse here. While on one hand it is more flexible by providing a configurable number of rounds, it requires a 8 byte nonce and only works on message sizes that are a multiple of 64 bytes. I plan on merging both implementations at some point though.

### Testing procedure

Execute the unittest on your favorite platforms and verify that the test doesn't fail. Furthermore check whether the documentation is sufficient for correct use.

### Issues/PRs references

depends on #9236, #10218 and #10219
